Homeworks/Homework W8/SQLQuery_HomeworkW8.sql

```diff
@@ -0,0 +1,340 @@
+CREATE DATABASE ZooManagemnt;
+
+USE ZooManagemnt;
+-- Creating tables
+CREATE TABLE ZoneArea
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Zone VARCHAR(50)
+);
+
+CREATE TABLE Species
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Name VARCHAR(50),
+Zone INT REFERENCES ZoneArea(Id)
+);
+
+CREATE TABLE Animals
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Species INT REFERENCES Species(Id)
+);
+
+CREATE TABLE Workers
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+FirstName VARCHAR (20),
+LastName VARCHAR (20),
+WorkArea INT REFERENCES ZoneArea(Id)
+);
+
+CREATE TABLE Ticket
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Price MONEY,
+ValidityStart DATE,
+ValidityEnd DATE,
+Zone INT REFERENCES ZoneArea(Id)
+);
+
+SELECT *
+FROM Ticket;
+
+CREATE TABLE Visitors
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+FirstName VARCHAR (20),
+LastName VARCHAR (20),
+Ticket INT REFERENCES Ticket(Id)
+);
+-- Populating tables + Changes related to tables if needed
+INSERT INTO ZoneArea
+VALUES ('Fish');
+INSERT INTO ZoneArea
+VALUES ('Amphibians');
+INSERT INTO ZoneArea
+VALUES ('Reptiles');
+INSERT INTO ZoneArea
+VALUES ('Birds');
+INSERT INTO ZoneArea
+VALUES ('Mammals');
+
+SELECT *
+FROM ZoneArea;
+
+INSERT INTO Species
+VALUES ('Goldfish',1);
+INSERT INTO Species
+VALUES ('Angelfish',1);
+INSERT INTO Species
+VALUES ('Neon Tetra',1);
+INSERT INTO Species
+VALUES ('Catfish',1);
+INSERT INTO Species
+VALUES ('Frog',2);
+INSERT INTO Species
+VALUES ('Toad',2);
+INSERT INTO Species
+VALUES ('Salamander',2);
+INSERT INTO Species
+VALUES ('Newt',2);
+INSERT INTO Species
+VALUES ('Turtle',3);
+INSERT INTO Species
+VALUES ('Lizard',3);
+INSERT INTO Species
+VALUES ('Snake',3);
+INSERT INTO Species
+VALUES ('Peafowl',4);
+INSERT INTO Species
+VALUES ('Emu',4);
+INSERT INTO Species
+VALUES ('Parrot',4);
+INSERT INTO Species
+VALUES ('Duck',4);
+INSERT INTO Species
+VALUES ('Swan',4);
+INSERT INTO Species
+VALUES ('Tiger',5);
+INSERT INTO Species
+VALUES ('Gorilla',5);
+INSERT INTO Species
+VALUES ('Leopard',5);
+INSERT INTO Species
+VALUES ('Bear',5);
+
+
+
+SELECT *
+FROM Species;
+
+SELECT *
+FROM ZoneArea;
+
+
+
+ALTER TABLE Workers
+DROP constraint FK__Workers__WorkAre__3F466844;
+ALTER TABLE Workers
+DROP COLUMN WorkArea;
+
+ALTER TABLE ZoneArea
+ADD Worker INT REFERENCES Workers(Id);
```

Comment:
It's a good practice to use the EntityNameId naming conventions for foreign keys. It makes the database easier to work with. So here it would have been WorkerId.